dask/bag/core.py

```diff
@@ -244,20 +280,39 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, **kwargs):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s).  Values in **kwargs must either be a non-dask python
+        literal or a dask.bag.Item.
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.  Example:
+
+        >>> b = db.from_sequence(range(5))
+        >>> b.map(
+        ...     lambda x, total=0: float(x) / total,
+        ...     total=b.sum()).sum().compute()
+        1.0
         """
-        name = 'map-{0}-{1}'.format(funcname(func), tokenize(self, func))
+        name = 'map-{0}-{1}'.format(funcname(func),
+                                    tokenize(self, func, kwargs))
         if takes_multiple_arguments(func):
             func = partial(apply, func)
-        dsk = dict(((name, i), (reify, (map, func, (self.name, i))))
+        dsk = self.dask.copy()
+        if kwargs:
+            dsk, kw_pairs = merge_kwargs(dsk, kwargs)
+            func = (with_kwargs, func, (dict, kw_pairs))
```

Comment:
Hmm, it may well be, I just didn't trust it up against the existing partial-fuscation of func in context, I'll see if that works locally :-)